### DISCUSSION TITLE
KeyError: 10? Can't run League() for hockey

### DISCUSSION THREAD

**Discussion by jwinik** [cwendt94/espn-api#322] [ESPN API]
Is anyone having an issue running the League() function? It looks like the error is in player.py, specifically for hockey.

`File "C:\Users...\site-packages\espn_api\hockey\player.py", line 12, in <listcomp>
    self.eligibleSlots = [POSITION_MAP[pos] for pos in json_parsing(data, 'eligibleSlots')]`

`KeyError: 10`

I'm wondering if anyone has had similar issues recently. I ran it all two days ago and it worked fine, but maybe I did something. Thanks in advance!

**Comment by cwendt94** [comment]
I just tested out initializing the `League` with my hockey league and didn't run into any errors. I will look into the code to better handle key error

**Reply by jwinik** [reply]
Thanks for looking into it. I put it away for a while and came back to it. I tried changing the year to 2021 and it worked fine. When you tried it, was it for 2022?

**Comment by jwinik** [comment]
So I think the problem is that in `POSITION_MAP` in `constant.py`, the keys in the dictionary go from 1-8. I would guess that there is now (for some reason) a 10 that is not allowing the functions in `player.py` to map the player position when the key is 10.

**Reply by cwendt94** [reply]
I couldn't find out what position `10` is from my league. I wonder if it is IR or another injury slot. I did update the logic to handle unknown positions though. The error should hopefully be fixed in package version `0.19.3`

**Reply by jwinik** [reply]
Just seeing this, thank you! I ended up adding 10 to that dictionary to work around it. But all works well now with the update. Thanks again!

**Comment by D3spina** [comment]
Hi !
I wasn't sure if I should start a new thread but I have a similar problem. I can't access my league created in 2022. I tried in debug but nothing prints. If I change the year to 2021, I have a request, however it does not find my league.

**Reply by jwinik** [reply]
Is it related to playoffs? Having trouble with the matchup class depending on which week I choose

**Reply by D3spina** [reply]
Maybe. Actually we are in playoffs

**Reply by cwendt94** [reply]
Could you post the error you are seeing? I am not seeing any issues when I load my 2022 hockey league.

**Reply by D3spina** [reply]
No error, just nothing in the terminal
Edit : new try after ending, now working.
